File: src/main/java/net/scriptgate/conway/common/Color.java
package net.scriptgate.conway.common;

public class Color extends ColorRGBA {

    public static final Color RED = new Color(255, 0, 0);
    public static final Color GREEN = new Color(0, 255, 0);
    public static final Color BLUE = new Color(0, 0, 255);
    public static final Color WHITE = new Color(255, 255, 255);
    public static final Color BLACK = new Color(0, 0, 0);
    public int r, g, b, a;

    public Color() {
    }

    public Color(int r, int g, int b) {
        this(r, g, b, 255);
    }

    public Color(int r, int g, int b, int a) {
        this.r = r;
        this.g = g;
        this.b = b;
        this.a = a;
    }

    @Override
    public int intValue() {
        return a << 24 | (r & 0xFF) << 16 | (g & 0xFF) << 8 | b & 0xFF;
    }
}

File: src/main/java/net/scriptgate/conway/common/Color.kt
package net.scriptgate.conway.common

class Color(var r: Int, var g: Int, var b: Int, var a: Int = 255) {


    fun intValue(): Int {
        return a shl 24 or (r and 0xFF shl 16) or (g and 0xFF shl 8) or (b and 0xFF)
    }

    companion object {

        val RED = Color(255, 0, 0)
        val GREEN = Color(0, 255, 0)
        val BLUE = Color(0, 0, 255)
        val WHITE = Color(255, 255, 255)
        val BLACK = Color(0, 0, 0)
    }
}
